marshal.go

```diff
@@ -444,18 +426,18 @@ func (dm *DependencyMap) extract(m map[string]interface{}) error {
 	dm.Names = make(map[string][]string)
 	dm.Schemas = make(map[string]*Schema)
 	for k, p := range m {
-		switch p.(type) {
+		switch p := p.(type) {
```

Comment:
I hear ya about the shadowing thing. I used to not liek out, too, though I'm slowly changing my mind.  I mostly did it here to make the diff cleaner.  I can fix it though.